newm-server/src/main/kotlin/io/projectnewm/server/pugins/auth/google/OAuthGoogle.kt

```diff
@@ -0,0 +1,56 @@
+package io.projectnewm.server.pugins.auth.google
+
+import io.ktor.client.HttpClient
+import io.ktor.client.engine.cio.CIO
+import io.ktor.http.Cookie
+import io.ktor.http.HttpMethod
+import io.ktor.server.application.ApplicationEnvironment
+import io.ktor.server.application.call
+import io.ktor.server.auth.Authentication
+import io.ktor.server.auth.OAuthAccessTokenResponse
+import io.ktor.server.auth.OAuthServerSettings
+import io.ktor.server.auth.authenticate
+import io.ktor.server.auth.oauth
+import io.ktor.server.auth.principal
+import io.ktor.server.response.respondRedirect
+import io.ktor.server.routing.Routing
+import io.ktor.server.routing.get
+import io.projectnewm.server.ext.getConfigInt
+import io.projectnewm.server.ext.getConfigString
+import io.projectnewm.server.ext.getConfigStrings
+
+fun Authentication.Configuration.configureOAuthGoogle(environment: ApplicationEnvironment) {
+    val port = environment.getConfigInt("ktor.deployment.port")
+    oauth("auth-oauth-google") {
+        urlProvider = { "http://localhost:$port/callback" }
```

Comment:
We need to update this with the current public URL of the server so it works anywhere

Reply:
That may be challenging, but I wonder if ktor already exposes this?